### PR TITLE
Users

### PR DESCRIPTION
**Pull request by RadetskiyBogdan**

### CODE REVIEW

File: src/controllers/users.js [hunk 1]
```diff
@@ -75,54 +75,60 @@ class UsersController {
...
-                    .status(404)
-                    .json({ success: false, error: `User not found` });
-            }
-
-            return res.status(200).json({ success: true, data: user });
-        })
-            .clone()
-            .catch((err) => console.log(err));
+        try {
+            await User.findOneAndDelete({ _id: req.params.id }, (err, user) => {
```
Comment (MaxLogvyniukFive) [inline src/controllers/users.js:79]: Move it to service

File: src/controllers/users.js [hunk 2]
```diff
@@ -75,54 +75,60 @@ class UsersController {
...
-                return res
-                    .status(404)
-                    .json({ success: false, error: `User not found` });
-            }
-            return res.status(200).json({ success: true, data: user });
-        })
-            .clone()
-            .catch((err) => console.log(err));
+        try {
+            await User.findOne({ _id: req.params.id }, (err, user) => {
```
Comment (MaxLogvyniukFive) [inline src/controllers/users.js:99]: Move it to service

File: src/controllers/users.js [hunk 3]
```diff
@@ -75,54 +75,60 @@ class UsersController {
...
-                return res
-                    .status(404)
-                    .json({ success: false, error: `User not found` });
-            }
-            return res.status(200).json({ success: true, data: users });
-        })
-            .clone()
-            .catch((err) => console.log(err));
+        try {
+            await User.find({}, (err, users) => {
```
Comment (MaxLogvyniukFive) [inline src/controllers/users.js:118]: Same

**Review by MaxLogvyniukFive — CHANGES_REQUESTED** [review]

File: src/controllers/auth.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ const { SessionService } = require('../services/session');
 
 class AuthController {
     static async registraition(req, res) {
-        let { name, email, password } = req.body;
+        let { name, email, lastName, password } = req.body;
```
Comment (MaxLogvyniukFive) [inline src/controllers/auth.js:14]: Use const

File: src/controllers/test.txt [hunk 5]
```diff
@@ -0,0 +1,192 @@
+const Movie = require('../models/movie');
```
Comment (MaxLogvyniukFive) [inline src/controllers/test.txt]: ?

**Review by MaxLogvyniukFive — CHANGES_REQUESTED** [review]

File: src/controllers/test.txt (hunk 5 above)
Comment (RadetskiyBogdan) [reply src/controllers/test.txt]: accidentally added my notes. already deleted

File: src/controllers/users.js [hunk 6]
```diff
@@ -1,120 +1,100 @@
...
+            return res.status(200).json(result);
+        } catch (err) {
+            return res.status(500).json({ errors: err });
+        }
     }
 
     static async deleteUser(req, res) {
         try {
-            const user = await User.findOneAndDelete({});
+            const user = await UserService.findById(req.params.id);
```
Comment (MaxLogvyniukFive) [inline src/controllers/users.js:53]: Extract "req.params.id" in const and if not received return appropriate server responce

File: src/controllers/users.js [hunk 7]
```diff
@@ -1,120 +1,100 @@
...
         } catch (err) {
-            res.status(500).json({ errors: err });
+            return res.status(500).json({ errors: err });
         }
     }
 
     static async getUserById(req, res) {
         try {
-            const user = await User.findOne({});
+            const user = await UserService.findById(req.params.id);
```
Comment (MaxLogvyniukFive) [inline src/controllers/users.js:72]: Same here

**Review by MaxLogvyniukFive — CHANGES_REQUESTED** [review]

**Pull request merged**